polyfill/test/regex.mjs

```diff
@@ -492,18 +492,19 @@ describe('fromString regex', () => {
       []
     );
 
-    day.forEach(([p, expect]) => {
+    day.slice(1).forEach(([p, expect]) => {
       test(`P${p}`, expect);
       test(`p${p}`, expect);
       test(`p${p.toLowerCase()}`, expect);
     });
-    tim.forEach(([p, expect]) => {
+    tim.slice(1).forEach(([p, expect]) => {
       test(`PT${p}`, expect);
       test(`Pt${p}`, expect);
       test(`pt${p.toLowerCase()}`, expect);
     });
     for (let [d, dexpect] of day) {
       for (let [t, texpect] of tim) {
+        if (d === '' && t === '') continue;
```

Comment:
This allows through empty `t` and then uses it to build input like "P3DT", which should instead be "P3D".

Reply:
+1 https://tools.ietf.org/html/rfc3339#appendix-A